Index: model/WaysNavigableLinkAdas.py
===================================================================
--- model/WaysNavigableLinkAdas.py	(revision 503667)
+++ model/WaysNavigableLinkAdas.py	(working copy)
@@ -293,7 +293,7 @@
             self.__count("%s%s"%(GLOBAL_KEY_PREFIX,keys[0]))
         elif (self.isNotEmpty(line[25]) and '0' != line[25] and '998' != line[25]) and ((not self.isNotEmpty(line[26])) or '0' == line[26] or '998' == line[26]):
             self.__count("%s%s"%(GLOBAL_KEY_PREFIX,keys[0]))
-        elif ((not self.isNotEmpty(line[25])) or '0' == line[25] or '998' == line[25]) and ((not self.isNotEmpty(line[26])) and '0' != line[26] and '998' != line[26]):
+        elif ((not self.isNotEmpty(line[25])) or '0' == line[25] or '998' == line[25]) and (self.isNotEmpty(line[26]) and '0' != line[26] and '998' != line[26]):
             self.__count("%s%s"%(GLOBAL_KEY_PREFIX,keys[0]))
 
     def __get_adas_special_maxspeed(self,keys,line):
